client/proposer/service.go

```diff
@@ -25,7 +25,7 @@ import (
 // sharding/service.go.
 type Proposer struct {
 	config    *params.Config
-	client    *mainchain.SMCClient
+	client    mainchain.FullClient
```

Comment:
Is there a reason you are not using `*`?

Reply:
It's an interface

Reply:
Oops. My bad for missing that.